src/compile/mark/encode/position-rect.ts

```diff
@@ -156,7 +162,8 @@ function positionAndSize(
     If band is 0.6, the the x/y position in such case should be `(1 - band) / 2` = 0.2
    */
 
-  const defaultBandAlign = scale?.get('type') !== 'band' || !('band' in sizeMixins[vgSizeChannel]) ? 'middle' : 'top';
+  const defaultBandAlign =
+    scale?.get('type') === 'band' && isRelativeBandSize(bandSize) && !hasFixedSizeMixins ? 'top' : 'middle';
```

Comment:
Can you explain if 
`('band' in sizeMixins[vgSizeChannel])` is equivalent `isRelativeBandSize(bandSize) && !hasFixedSizeMixins`?

I wonder if this logic change is correct and necessary

Reply:
Yeah, since we output `{signal: max(0.25, bandwidth('x'))}`  instead of `{"scale": "x", "band": 1}`, the old logic that checks for "band" doesn't work anymore. 

I tracked how `band` ` in `sizeMixins[vgSizeChannel]` was generated and found that  

`band` in `sizeMixins[vgSizeChannel]` only happens if the "bandSize" is relative and there is no size from mark/encoding (i'll rename the variable to `hasSizeFromMarkOrEncoding`).